File: benchmark/javascript/task6_fileio.js
const fs = require('fs');

function countWords(inputFile, outputFile) {
    const text = fs.readFileSync(inputFile, 'utf8');
    const counts = {};

    text.split(/\s+/).filter(w => w).forEach(word => {
        counts[word] = (counts[word] || 0) + 1;
    });

    const output = Object.entries(counts)
        .map(([word, count]) => `${word}: ${count}`)
        .join('\n');

    fs.writeFileSync(outputFile, output);
}

File: benchmark/dart/task6_fileio.dart
import 'dart:io';

void countWords(String inputFile, String outputFile) {
  var text = File(inputFile).readAsStringSync();
  var counts = <String, int>{};

  for (var word in text.split(RegExp(r'\s+'))) {
    if (word.isNotEmpty) {
      counts[word] = (counts[word] ?? 0) + 1;
    }
  }

  var output = counts.entries.map((e) => '${e.key}: ${e.value}').join('\n');
  File(outputFile).writeAsStringSync(output);
}
